src/averageAges.js

```diff
@@ -55,7 +65,16 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const women = people.filter(woman => woman.sex === 'f'
+  && people.some(child => child.mother === woman.name));
+  const children = people.filter(man => onlyWithSon ? man.sex === 'm'
+  && people.some(mother => man.mother === mother.name)
```

Comment:
In 71 string I would change variable 'mother' by something less confused. for example "person", cose var "woman" you have already & "mother" also yse as key in {objects}